multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -309,51 +309,62 @@ func (c *Cluster) validateKeyPair(ec2Svc ec2Service) error {
 type r53Service interface {
 	ListHostedZonesByName(*route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesByNameOutput, error)
 	ListResourceRecordSets(*route53.ListResourceRecordSetsInput) (*route53.ListResourceRecordSetsOutput, error)
+	GetHostedZone(*route53.GetHostedZoneInput) (*route53.GetHostedZoneOutput, error)
 }
 
 func (c *Cluster) validateDNSConfig(r53 r53Service) error {
 	if !c.CreateRecordSet {
 		return nil
 	}
 
-	listHostedZoneInput := route53.ListHostedZonesByNameInput{
-		DNSName: aws.String(c.HostedZone),
-	}
+	if c.HostedZoneID == "" {
+		//TODO(colhom): When HostedZone parameter is gone, this block can be removed
+		//Config will gaurantee that HostedZoneID is set from the get-go
+		listHostedZoneInput := route53.ListHostedZonesByNameInput{
+			DNSName: aws.String(c.HostedZone),
+		}
 
-	zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
-	if err != nil {
-		return fmt.Errorf("Error validating HostedZone: %s", err)
-	}
+		zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
+		if err != nil {
+			return fmt.Errorf("Error validating HostedZone: %s", err)
+		}
 
-	zones := zonesResp.HostedZones
+		zones := zonesResp.HostedZones
 
-	if len(zones) == 0 {
-		return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
-	}
+		if len(zones) == 0 {
+			return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
+		}
 
-	var matchingZone *route53.HostedZone = nil
-	for _, zone := range zones {
-		if aws.StringValue(zone.Name) == c.HostedZone {
-			//TODO:(chom) for v1, we will require HostedZoneID be
-			//set, as HostedZone will be going away
-			if c.HostedZoneID == "" || aws.StringValue(zone.Id) == c.HostedZoneID {
+		var matchingZone *route53.HostedZone = nil
+		for _, zone := range zones {
+			if aws.StringValue(zone.Name) == c.HostedZone {
 				matchingZone = zone
 				break
 			}
 		}
-	}
-	if matchingZone == nil {
-		if c.HostedZoneID == "" {
+		if matchingZone == nil {
 			return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
 		}
-		return fmt.Errorf("hosted zone with id %s, name %s does not exist", c.HostedZoneID, c.HostedZone)
+		c.HostedZoneID = aws.StringValue(matchingZone.Id)
+	}
+
+	hzOut, err := r53.GetHostedZone(&route53.GetHostedZoneInput{Id: aws.String(c.HostedZoneID)})
+	if err != nil {
+		return fmt.Errorf("error getting hosted zone %s: %v", c.HostedZoneID, err)
+	}
+
+	if !isSubdomain(c.ExternalDNSName, aws.StringValue(hzOut.HostedZone.Name)) {
+		return fmt.Errorf("externalDNSName %s is not a sub-domain of hosted-zone %s", c.ExternalDNSName, aws.StringValue(hzOut.HostedZone.Name))
```

Comment:
I guess this would catch a potential mismatch of "first hosted zone found" from comment above. But do we want this implicit functionality? e.g. this could work for a while, then it just so happens that it stops working because the hostedzone is no longer the first (and therefor matching) zone.

I'll defer to you on this. I'm not sure if order is fixed or if we should just hard-fail on multiple.